tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -4148,6 +4155,24 @@ static __always_inline bool skb_revalidate_data(struct __sk_buff *skb, uint8_t *
     return true;
 }
 
+static __always_inline bool is_dns_request(uint8_t * head, uint8_t* tail, struct __sk_buff * skb, net_packet_t * pkt){
+    if(pkt->dst_port == 53)
+        return true;
+    return false;
+}
+
+/* check_protocols is checking for potential network protocols based packets
+ * if it is finds a potential protocol based packet it sends it to the user-space with the eventID
+ * in the user-space we checking the packet data more deeply to verify it is the acual protocl,
+ * and if it is- we parsing it and alerting to the user space
+*/
+static __always_inline void check_protocols(struct __sk_buff *skb, uint8_t* head, uint8_t* tail, struct bpf_map_def *events_channel, net_packet_t* pkt, u64 flags){
+   if (is_dns_request(head, tail, skb, pkt) && event_chosen(NET_DNS_REQUEST)){
+        pkt->event_id = NET_DNS_REQUEST;
+        bpf_perf_event_output(skb, &net_events, flags, pkt, sizeof(net_packet_t));
```

Comment:
in this case we send the packet twice - once to capture the packet (save it in the pcap), and once to parse DNS values from it. 
you created generic handlers from network events, so i think we can send the packet only once.
if it is recognized as DNS packet here in the bpf, then change the `pkt.event_id` to `NET_DNS_REQUEST`, if not, then the `pkt.event_id` should stay as it is (`NET_PACKET`).

then - in `ProcessNetEvent()` you should return `true` in the case of `NetDnsRequest`.

let me know if that could work.

Reply:
sound good, ill try it! (:

Reply:
it Works, after ill fixed your other comments ill push it too